code/datums/stars.dm

```diff
@@ -227,31 +227,27 @@
 	var/primary_resource
 	var/is_primary = 0
 
+	var/datum/station_module/module
 
 /datum/space_station/New(var/datum/planet/P)
 	planet = P
 	SSstarmap.stations += src
 
-/datum/space_station/proc/generate()
-	// TODO: Implement a more sophisticated way of generating station stocks.
-
-	stock[SSshuttle.supply_packs[/datum/supply_pack/munitions/he]] = rand(1,10)
-	if(prob(33))
-		stock[SSshuttle.supply_packs[/datum/supply_pack/munitions/sp]] = rand(1,10)
-	else if(prob(50))
-		stock[SSshuttle.supply_packs[/datum/supply_pack/munitions/sh]] = rand(1,10)
 
-	stock[SSshuttle.supply_packs[/datum/supply_pack/gas/o2]] = rand(1,8)
-	stock[SSshuttle.supply_packs[/datum/supply_pack/gas/n2]] = rand(1,2)
-	stock[SSshuttle.supply_packs[/datum/supply_pack/gas/plasma]] = rand(1,5)
-	stock[SSshuttle.supply_packs[/datum/supply_pack/misc/space_yellow_pages]] = rand(1,5)
+/datum/space_station/proc/generate(var/list/module_weights)
+	// TODO: Implement a more sophisticated way of generating station stocks.
+	for(var/I in SSshuttle.supply_packs)
+		var/datum/supply_pack/pack = SSshuttle.supply_packs[I]
+		var/probability = pack.base_chance_to_spawn
+		for(var/keyword in module.buy_keywords)
+			if(pack.chance_modifiers && keyword in pack.chance_modifiers)
+				probability = pack.base_chance_to_spawn + pack.chance_modifiers[keyword]
+		if(prob(probability))
+			if(pack.min_amount_to_stock != -1 && pack.max_amount_to_stock)
+				stock[I] = rand(pack.min_amount_to_stock, pack.max_amount_to_stock)
+			else
+				stock[I] = pack.min_amount_to_stock
```

Comment:
put a CHECK_TICK in here

Reply:
check tick exists in latest commit, git's being wonky and doesn't want to update this comment